java/ql/src/experimental/CWE-532/SensitiveInfoLog.ql

```diff
@@ -31,8 +31,10 @@ class CredentialExpr extends Expr {
 class LoggerType extends RefType {
   LoggerType() {
     this.hasQualifiedName("org.apache.log4j", "Category") or //Log4J
+    this.hasQualifiedName("org.apache.logging.log4j", "Logger") or //Log4j 2
     this.hasQualifiedName("org.slf4j", "Logger") or //SLF4j and Gradle Logging
-    this.hasQualifiedName("org.jboss.logging", "BasicLogger") //JBoss Logging
+    this.hasQualifiedName("org.jboss.logging", "Logger") or //JBoss Logging
```

Comment:
This should probably be [`BasicLogger`](https://docs.jboss.org/jbosslogging/latest/org/jboss/logging/BasicLogger.html).

It appears there is also [`org.jboss.logging.Logger`](https://docs.jboss.org/jbossas/javadoc/4.0.4/common/org/jboss/logging/Logger.html) which was part of "JBoss Application Server" (now "WildFly"), however I could not find it in "WildFly" yet, so it might have been removed.

There is also [`org.jboss.logging.Logger`](https://docs.jboss.org/jbosslogging/latest/org/jboss/logging/Logger.html) in JBoss Logging, however that class implements `BasicLogger`.

Reply:
Hi @Marcono1234,

I intentionally changed `org.jboss.logging.BasicLogger` to `org.jboss.logging.Logger`. The rationale is that `BasicLogger` is an interface and the typical usage is with `Logger`. For example, the following is shown in the [JBoss Logging guide](https://access.redhat.com/documentation/en-us/jboss_enterprise_application_platform/6/html/development_guide/add_logging_to_an_application_with_jboss_logging):
> private static final Logger LOGGER = Logger.getLogger(HelloWorld.class);

The `Logger` class is indeed in the JBoss [implementation](https://github.com/jboss-logging/jboss-logging/tree/master/src/main/java/org/jboss/logging). For the latest WildFly server, the class is under the path 
`⁨wildfly-20.0.0.Final⁩ ▸ ⁨modules⁩ ▸ ⁨system⁩ ▸ ⁨layers⁩ ▸ ⁨base⁩ ▸ ⁨org⁩ ▸ ⁨jboss⁩ ▸ ⁨logging⁩ ▸ ⁨main⁩ ▸ ⁨jboss-logging-3.4.1.Final⁩ ▸ ⁨org⁩ ▸ ⁨jboss⁩ ▸ ⁨logging⁩`.

Thanks for your review and hopefully this helps to clarify.

@luchua-bc

Reply:
Thanks for explaining your rationale.
Though it would probably be still good to match `BasicLogger` since there might be other subtypes which do not extend the `Logger` class, see https://docs.jboss.org/jbossas/javadoc/7.1.2.Final/org/jboss/logging/BasicLogger.html which lists some interfaces and classes extending it.

Regarding my "JBoss Application Server" comment, I meant that the [`org.jboss.logging.Logger`](https://docs.jboss.org/jbossas/javadoc/4.0.4/common/org/jboss/logging/Logger.html) in JBoss Application Server 4.0.4 did not implement `BasicLogger`, though it appears that `Logger` class does not exist anymore and the one you found implements `BasicLogger`.

Reply:
Thanks @Marcono1234. I've made the change to check both `BasicLogger` and `Logger` of JBoss Logging, which covers both the case that `Logger` extends `BasicLogger` and the case that it doesn't in some implementations like the one you mentioned. I've added comments to make it clear.